src/topojson/topojson.ml

```diff
@@ -248,6 +248,7 @@ module Make (J : Intf.Json) = struct
       geometry : geometry;
       properties : properties;
       foreign_members : (string * json) list;
+      id : id;
```

Comment:
Adding an `id` field to this record seems like a good choice but there are some issues.

1. You have given it the type `id` but no such type exists. It probably should be of type `json` and it should also be optional. The spec doesn't _require_ and `id` field.
2. Anywhere we subsequently construct a geometry record needs to be update (e.g. in the function `v` directly afterwards)
3. The JSON encoders (e.g. `Geometry. base_of_json`) will need updating too.

Reply:
@patricoferris Thank you for clarifying me
1. Line 251 will be replaced with id : json;
2. line 254 to 259  will become
let v ?(properties = `None) ?(foreign_members = []) ?(id={}) geo =
      { geometry = geo; properties; foreign_members }

    let geometry t = t.geometry
    let properties t = t.properties
    let foreign_members t = t.foreign_members
    let id t = t.id

Please correct me and please which command will I use to compile locally before making a PR. Thank you

Reply:
Great :)) Note that in general I'm not telling you _everything_ you should or shouldn't do as it's good for you to discover these things yourself as well, and in general I'll try to link to documentation rather than tell you directly to help you discover answers yourself. Please don't hesitate to ask questions though!

Re (1) -- It should probably be optional, OCaml has optional types. See https://cs3110.github.io/textbook/chapters/data/options.html. This should also help with (2). From your code (you can format code with [codeblocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) in Github) it looks like the record is still missing it's `id` value.

See the contributing guide in the root of this repo for how to build the code.

Reply:
@patricoferris Please can you reply 👆 Thank you